amazon_cloud_code_generator/cmd/refresh_modules.py

```diff
@@ -122,27 +122,61 @@ def generate_params(definitions: Iterable) -> str:
     return params
 
 
+def gen_mutually_exclusive(schema: Dict) -> List:
+    primary_idenfifier = schema.get("primaryIdentifier", [])
+    entries: List = []
+
+    if len(primary_idenfifier) > 1:
+        entries.append([tuple(primary_idenfifier), "identifier"])
+
+    return entries
+
+
 def gen_required_if(schema: Dict) -> List:
     primary_idenfifier = schema.get("primaryIdentifier", [])
     required = schema.get("required", [])
     entries: List = []
     states = ["absent", "get"]
 
+    _primary_idenfifier = copy.copy(primary_idenfifier)
+
+    # For compound primary identifiers consisting of multiple resource properties strung together,
+    # use the property values in the order that they are specified in the primary identifier definition
+    if len(primary_idenfifier) > 1:
+        entries.append(["state", "list", list(set(primary_idenfifier[:-1])), True])
```

Comment:
I think this is fine, because in this case we don't actually care what order these are in. If I'm wrong and we do care, then we can't use `set()` here because sets in python are unordered data structures.

Reply:
We only care when we do primary_idenfifier[:-1]. But I removed the useless casts.